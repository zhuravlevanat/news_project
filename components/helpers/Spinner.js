import { Box } from "@material-ui/core";
import LinearProgress from "@material-ui/core/LinearProgress";
import React from "react";

const Spinner = () => (
    <Box  height='100vh' pb={25} display={"flex"} flexDirection={"column"} justifyContent={"center"}>
        <LinearProgress color="primary"/>
        <LinearProgress color="primary"/>
    </Box>
);

export default  Spinner;