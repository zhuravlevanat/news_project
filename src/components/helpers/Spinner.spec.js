import React from 'react';
import Enzyme, { shallow, mount } from 'enzyme';
import toJson from 'enzyme-to-json';
import Adapter from 'enzyme-adapter-react-16';
import Spinner from "./Spinner";

Enzyme.configure({ adapter: new Adapter() });

describe('Spinner', () => {
    describe('Spinner box', () => {
        const spinner = shallow(<Spinner/>);
        it('renders without crashing', () => {
            expect(toJson(spinner)).toMatchSnapshot();
        });
        it('renders 1 box', () => {
            expect(spinner.find('Styled(MuiBox)'))
                .toHaveLength(1);
        });
    });
    describe('Spinners', () => {
        const spinner = mount(<Spinner />);
        it('renders container 2 spinner', () => {
            expect(spinner.find('WithStyles(ForwardRef(LinearProgress))')).toHaveLength(2);
        });
    });
});
