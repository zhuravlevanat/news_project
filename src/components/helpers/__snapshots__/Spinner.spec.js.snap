// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Spinner Spinner box renders without crashing 1`] = `
<Styled(MuiBox)
  display="flex"
  flexDirection="column"
  height="100vh"
  justifyContent="center"
  pb={25}
>
  <WithStyles(ForwardRef(LinearProgress))
    color="primary"
  />
  <WithStyles(ForwardRef(LinearProgress))
    color="primary"
  />
</Styled(MuiBox)>
`;
